odoo/addons/base/models/res_lang.py

```diff
@@ -216,9 +218,11 @@ def _lang_get_id(self, code):
     def _lang_get_direction(self, code):
         return self.with_context(active_test=True).search([('code', '=', code)]).direction
 
-    @tools.ormcache('url_code')
     def _lang_get_code(self, url_code):
-        return self.with_context(active_test=True).search([('url_code', '=', url_code)]).code or url_code
+        for lg in self.get_available():
+            if lg[3] and (lg[1] == url_code):
+                return lg[0]
```

Comment:
Could we please name those instead of using the index? It's really hard to figure without going through the stack to see which one is 3, which one is 1 etc.
```suggestion
        for code, url_code, name, active, *_ in self.get_available():
            if active and (url_code == url_code_param):
                return code
```